core/client/fs/src/main/java/alluxio/client/block/stream/DefaultBlockWorkerClient.java

```diff
@@ -217,23 +216,11 @@ public ClearMetricsResponse clearMetrics(ClearMetricsRequest request) {
   public void cache(CacheRequest request) {
     boolean async = request.getAsync();
     if (async) {
```

Comment:
I thought it would worsen readability if we get rid of this if-else since the exception handling is quite different. I update the code a little bit to make these two paths more clear. let me know if you have a better idea.

Reply:
I don't see any exception handling for sync case here. Why do we handle for one case only?

Reply:
IMO, it is better to have unified try-catch.
error handling for async cache should not be in this `DefaultBlockWorkerClient` but on the caller side (e.g., these InStreams). Important logic should not be put here as it gets easily ignored.